Index: /work/boston/mono/mono/mini/inssel.brg
===================================================================
--- /work/boston/mono/mono/mini/inssel.brg	(revision 112046)
+++ /work/boston/mono/mono/mini/inssel.brg	(working copy)
@@ -779,6 +779,7 @@
 }
 
 stmt: OP_NOP "0" {
+	mono_bblock_add_inst (s->cbb, tree);
 }
 
 stmt: OP_BREAK "0" {
Index: /work/boston/mono/mono/mini/mini-amd64.c
===================================================================
--- /work/boston/mono/mono/mini/mini-amd64.c	(revision 112046)
+++ /work/boston/mono/mono/mini/mini-amd64.c	(working copy)
@@ -2668,6 +2668,8 @@
 			amd64_breakpoint (code);
 			break;
 		case OP_NOP:
+			amd64_nop (code);
+			break;
 		case OP_DUMMY_USE:
 		case OP_DUMMY_STORE:
 		case OP_NOT_REACHED:
Index: /work/boston/mono/mono/mini/cpu-amd64.md
===================================================================
--- /work/boston/mono/mono/mini/cpu-amd64.md	(revision 112046)
+++ /work/boston/mono/mono/mini/cpu-amd64.md	(working copy)
@@ -54,6 +54,7 @@
 # See the code in mini-x86.c for more details on how the specifiers are used.
 #
 break: len:2
+nop: len:2
 jmp: len:120
 br: len:6
 label: len:0
